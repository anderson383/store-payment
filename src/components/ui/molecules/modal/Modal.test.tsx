
import {
  render,
  screen
} from '@testing-library/react';
import { act } from 'react';
import Modal from './modal';
import { createRoot } from 'react-dom/client';
describe('Tests in Modal component', () => {

  const setOpenModal = jest.fn()

  let container:any = null;
  let root:any = null;

  afterEach(() => {
    act(() => {
      root.unmount();
    })
    container?.remove();
    container = null;
  });

  beforeEach(() => {
    container = document.createElement('div');
    container.setAttribute('id', 'radix-:r0:');
    container.setAttribute('role', 'dialog');
    root = createRoot(container!);

    document.body.appendChild(container);
  });

  const setup = () => {
    return render(
      <Modal
        openModal={true}
        setOpenModal={setOpenModal}
      >
        <p>Render modal</p>
      </Modal>
    );
  };

  test('Should render the component', async () => {
    const {asFragment, debug} = setup();
    expect(screen.getByText('Render modal')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
