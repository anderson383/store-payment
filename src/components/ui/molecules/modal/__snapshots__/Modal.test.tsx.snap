// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Modal component Should render the component 1`] = `<DocumentFragment />`;
